sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MathExpressionsSuite.scala

```diff
@@ -669,19 +738,62 @@ class MathExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
       checkEvaluation(BRound(intPi, scale), intResultsB(i), EmptyRow)
       checkEvaluation(BRound(longPi, scale), longResults(i), EmptyRow)
       checkEvaluation(BRound(floatPi, scale), floatResults(i), EmptyRow)
+      checkEvaluation(checkDataTypeAndCast(
+        FloorExpressionBuilder.build(Seq(Literal(doublePi), Literal(scale)))),
```

Comment:
This is UT, we can test against `RoundCeil` directly.

Reply:
No we can't, because the casting for negative scale, int, short and floats are done at `FloorExpressionBuilder `